Alloy/Shaders/Shaders.metal

```diff
@@ -958,3 +951,63 @@ kernel void ycbcrToRGBA(texture2d<float, access::sample> sourceY [[ texture(0) ]
 
     destinationRGBA.write(destinationValue, position);
 }
+
+// MARK: - RGBA to YCbCr
+
+constant float4x4 rgbaToYCbCrTransform = {
+    { +0.2990f, -0.1687f, +0.5000f, +0.0000f },
+    { +0.5870f, -0.3313f, -0.4187f, +0.0000f },
+    { +0.1140f, +0.5000f, -0.0813f, +0.0000f },
+    { -0.0000f, +0.5000f, +0.5000f, +1.0000f }
+};
+
+kernel void rgbaToYCbCr(texture2d<float, access::sample> sourceRGBA [[ texture(0) ]],
+                        texture2d<float, access::write> destinationY [[ texture(1) ]],
+                        texture2d<float, access::write> destinationCbCr [[ texture(2) ]],
+                        const ushort2 position [[ thread_position_in_grid ]]) {
+    const auto textureSize = ushort2(destinationCbCr.get_width(),
+                                     destinationCbCr.get_height());
+    checkPosition(position, textureSize, deviceSupportsNonuniformThreadgroups);
+
+    if (halfSizedCbCr) {
+        constexpr sampler s(coord::pixel,
+                            address::clamp_to_edge,
+                            filter::nearest);
+        const auto positionF = float2(position);
```

Comment:
this approach is fine, but in this case we should have a CPU side assert on RGBA.size == Y.size